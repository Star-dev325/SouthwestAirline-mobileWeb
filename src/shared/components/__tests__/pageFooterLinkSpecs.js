import React from 'react';
import { shallow } from 'enzyme';
import PageFooterLink from 'src/shared/components/pageFooterLink';

describe('PageFooterLink', () => {
  context('default', () => {
    it('should render a browser link', () => {
      const pageFooterLink = createComponent({
        linkText: 'Link Text',
        linkType: 'browser',
        linkTarget: 'theUrl'
      });

      expect(pageFooterLink).toMatchSnapshot();
    });

    it('should render an app link', () => {
      const pageFooterLink = createComponent({
        linkText: 'Link Text',
        linkType: 'app',
        linkTarget: 'theUrl'
      });

      expect(pageFooterLink).toMatchSnapshot();
    });

    it('should render a webview link', () => {
      const pageFooterLink = createComponent({
        linkText: 'Link Text',
        linkType: 'webview',
        linkTarget: 'theUrl'
      });

      expect(pageFooterLink).toMatchSnapshot();
    });

    it('should render a link with a separator when showSeparator is true', () => {
      const pageFooterLink = createComponent({
        linkText: 'Link Text',
        linkType: 'browser',
        linkTarget: 'theUrl',
        showSeparator: true
      });

      expect(pageFooterLink).toMatchSnapshot();
    });
  });

  const createComponent = (props = {}) => shallow(<PageFooterLink {...props} />);
});
