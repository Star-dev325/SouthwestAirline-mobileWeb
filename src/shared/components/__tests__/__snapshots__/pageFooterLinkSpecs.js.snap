// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageFooterLink default should render a browser link 1`] = `
<div
  className="page-footer-wcm--link"
  key="link_text"
>
  <Connect(ContentLink)
    href="theUrl"
    id="link_text"
    target="_blank"
  >
    Link Text
  </Connect(ContentLink)>
</div>
`;

exports[`PageFooterLink default should render a link with a separator when showSeparator is true 1`] = `
<div
  className="page-footer-wcm--link"
  key="link_text"
>
  <Connect(ContentLink)
    href="theUrl"
    id="link_text"
    target="_blank"
  >
    Link Text
  </Connect(ContentLink)>
  <span
    className="page-footer-wcm--divider"
  >
    |
  </span>
</div>
`;

exports[`PageFooterLink default should render a webview link 1`] = `
<div
  className="page-footer-wcm--link"
  key="link_text"
>
  <Connect(ContentLink)
    href="theUrl"
    id="link_text"
    target="_self"
  >
    Link Text
  </Connect(ContentLink)>
</div>
`;

exports[`PageFooterLink default should render an app link 1`] = `
<div
  className="page-footer-wcm--link"
  key="link_text"
>
  <Connect(ContentLink)
    href="theUrl"
    id="link_text"
    target="_self"
  >
    Link Text
  </Connect(ContentLink)>
</div>
`;
